### PR TITLE
Add live code support

### PR DESCRIPTION
**Pull request by nihgwu**
- [ ] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code.
- [ ] Include the URL where we can test the change in the body of your PR.

This pull request:

- [ ] Fixes a bug
- [x] Adds additional features/functionality
- [ ] Updates documentation or example code
- [ ] Other

[Preview](https://radix-website-git-fork-nihgwu-neo-react-runner.modulz-deploys.com/docs/primitives/components/accordion)

Todos:

- [x] update mdx files, remove the embed example
- [x] create themes(dark/light) for editor following in previous style
- [x] bring back code highlight in `DocCodeBlock` as we removed `rehype-highlight-code`, as we need to consume code directly

This PR is a PoC to support live preview, not sure if that's really need though, [here is another example](https://play-react.vercel.app/#JYWwDg9gTgLgBAJQKYEMDG8BmUIjgcilQ3wG4AoUSWOAbzgGcYBPAGyQBMAaOAayWbYUIJAzgBfONlwEAAk2Aw0AC1EB6IuhhlK4aPHoA3YBHYweIFAFdDSHgCNW6XgEEJUnHnyyoKDsAAPAForYDU0U2gGHSp9OnI4OAAJYXsrKABzJCgAWSQAOysASQj8rgS4ABEIGAAxYFZ2DhKIMoqAYVU0Xha2xM6kQxx8hGAM5Rhe8slpLx8-QJCwzQwg4FLoiliaACo4FDFKnDAOCAB3fLzCgAUoUEVgWw8Zb19-YNCNYhggjmPTi5BESFHTkDbwBisYAcJAAVTALnyHFqfiQcAAvHwBEIRAwABS0Cr4AAMAFJ8AAuOhwCBgdCKZhU4k8GC+fIMTDQEBU-CslDspwwJAATTxACYwAEAJT4CTlRL4ACMxLJlOptPpLCpipZbI5XJ5fIFKCFouJMrl5HEUooYNaTEYUJho3GMERyNRGKxgl8uIJRNVVPoGrQDKZuv5+qg3IIRshJqQAA08UEJdLZeJ5QRlYH1XTQ1q4Dq4HHOdHDXrBUm8eaM9MbeQ7eyIU6kNULu6UTCvfwfcJRP6Fbng-mw3BmSW9WWY7zKwnRanJRbM0Sc+SgzTR4Xi6WDbG56aa8v67bwY7oUgADJITBupFdtGY3s4geEofrvOaxnjiPs6cVyMq2TNNj1XFUPxHL9tV-KMZzjICjzrK0GybB0AGUWCadpWiFfJ4ExJg2E4PEjlpAFLgKKxbnuGBHiQAA6bC8IKcx4kSEBgHyAB1aEYGUKkxTFZkKnsZwMhwKwkWw1hoB5M5lEUJB8CzexoBhKAEAWKwGCpAA2LM6Q4fx8gyKkAFYVIgAI0OUPxzgpCoh0lItiWcgBmAAOZygmVZzMlE8UxR4MU3OCgAWHhiXotyzKlSLnN8gI4CE7zFTMvyMgCwTgtC5KIvHeixRlLNvBEfwUDgPEwCITBsgYIIiA4Kw0E4IEahMfIqXyCAgmqm9sgKFqZSDRz9nyUATQ6yp0km1oeTClUQGiLNEn5CbaNaAAVe4TNqSSMA6nk0Csex1iCewkAAL2AbIa3oxV9KLSLop4RVitGs4GlYTp+SyAC-y5HgQwZZTRvwAAyABtDgTRQIImATdEACJaQKJGAF01TfRIcYh6HYfhi9kZgWkMaxsb1o6gA5fsqUhC923yTtPRXHHcahmGYDhhgiaRu5XTJjc1ssDb8hpkQ6dba9b2Z7tWbZggOYJnmYWR1SYBJkBBepYXZrF2nzxheFZbReW2bxznud59hb21+hddF8WkEli8XQmE3LTZ+WV2tU97XgRSQAwoixExbGUEaHlJIYJBtCzTk8LQ4BLudotQoqKF8iQJIkDGCZoIqCIZKgKljFMWP6LLsxFR1ES1OyTT-G0ql08SfwGDAJxv3wTB2ACUHVqhDJ8iKIUlqOljsgHuBVDzmABIsipDOM0yCGJOB0v7gyIB50WeSIQU6On5fOIyaX5+SxfEm07I0KQdgMB5bqs9Bokla535gAYFBHE4TGRrZkXWScBLA2AYqA2wHkVpwEgJxIUUAACitg8I6QIM-JSWYVxvwpJyY60QAE41Et0cSEBJIcGksAquFcqEwAAJzQKASXOANDK4mGrpg6YftmxwGDk0UeSA8AEUwsRUiJxzgURuHcDitFbD0X4SAHg9B6LKMDrw0QEgGxnjUeQrovBVIBHkV6QiTQSL-HEVcKiUiHiyIGN0fR8jFFwGUfRVRwixC+1QvAbRjcTCGKEURDgpiyLmMotRaRdF6I+IgA46kzjXEhw0RQLRwiOCbX5lkKAfjGApKCWIi4FiwnWIYmksYGSYnYwtsrLmQpkao3yHbUaRDeAkLIRQphLCaERVGow0ubDqF9JgDXDhFQ4ljzUQwE8jZkkBMvD-e+RicmiPIgUqxMiGKzIuqwRRS8-Ar3PgveOOEk4p21EFDOnFs651dDyMUm9p49JAdYWRECkBDOQlw9CKS750l8CTKACyAm5OWaE1ZETvkoF+dAbZiRZ7XKeiJMSEkpKRHaQM1h5cYD6QqJYTInFzKTM8TwlJ8iihInWCaaAAKTFLJCZImiESSVkrQBSqA0KYE7weHNAgP8GCmCsEKaeNsL7CUSJ9DgfEDkVHbp3FA3dOKZyQEEXuSAt4VAjmMEeY9UH4BasxKA08ABW2laKYGYExXCF8dWT31QS6ZTQXBQBwGcKlIizH5JBfS2RDqnVsswF9OSCkBUErUGoOACCAjUBgAwcgKrI1wDPDS91hQvSJokZYz1DEEAQBqBQWNcQE1urTSUjIGSU2FpWRm+ixaMm5ojfm-2VRy2UXNSxL02iW14VrXGgtwSk3FDHm24lY8u31u4amixti9FWSye23R9jh0xrrTQHteS01RIAOISTAGW3tabClrMiQsCAm7SFgBHcuht47KJRJnSkm9C680XrHU2wojL-DMr+YOgJb7yV-PPfAFdwLCjVuyLegJIHMkPqXQBy9L6rAbPmZibRCHWD-vjbB3dFjwWQv+Uhr5SAfksrQ4B2lVhvXnC-fax15xbQhrgMKUh-yUBgDAM4wlAAhKyLrAn4H8IYZSdAPGErdjANCMl8LZMBbxx4AnsY4oyJxfZ3L+UQGPrs0+SmhJZgeS8+iLy3kKhwRAPBcAAB8cBwbkwefgeSikkIKnxh-duP8mjows1ZlFVJdMvI8paITZ5ejsf5STfI3G8T4DSBrVosm1WRwINHWO08E51GEA0OV+RVB3DjnXKA6kfHN2zOBaesL85wBilmMVEqytXzgNKruPJ5UXKVX3ae6rh7yO1bq+BhrjXAFNR2y1XWp7ac88wtFNCDNwCaS05FxcA12cstZWyAIqQAAN15phcs5AAJLQRwzgXD0X290FwAB2cQq2SrgwpMoCAth9Ubmm0i8ho2OkDLcp7RWRm8Hk30TZOyZw1vr2B8lZyx3eCrb8yhR9MHn2YcopUAREAvR4ilBicz2MzyQ1UhAXgOLeAMEnZwHgMcYCcdx-jwnui-5emQFoeiN8MIJjxKyKwSAGyJCx+kVgVOkDdGJ4wWOsIoA86JxwNzmI6cYAZzHJnQo8SYAjjHDn6HuGQzAHVVoJPY7XE1-kCXiBvgy6QHLpA4WNd-FUyhRIRAYDpFC3iUaAAeTjARTOjUSE7q9hR3cKxxl7uDEH9hU4aBwX3fv-eBeC60fYdw4ZOE2cjdoxrcBfzRLSUWDAkbh4j-7lIIA0iZGyBY3ocA1A54j07tQUeov5Ar2zKv3urAQfdx7-3TeBuMAvAAeUwJgUn6JaBmXEPXhWAf4evrHqPv3VMkDOs2j-OATuRNiZqKZwAGMQAGpNpV5X+J6f-u1BN-kQfz3x+p9t5n3PuAPEkQUeX3PVfMAN+b6prvx-+-L+H-PwI0-S+f94DOa-xh5f5syz7OrUSGAJg36cQAhL575r6ADnhJvlvm-moAgc-qAUvkfnBiflgePquhYn-mfoHukqBhfrnn7jkNAGiJtNmjzlgQ3hgcQWPgMEMK0CJqXuXowYfswTwY3qQaUuQb-vwR3jhK2irEgL3v3rHIPmKJIG1vkNIQPrQEEMPiwSQRPv2iIZQRHmhCgE8NcCgFkHAC4AwIAGQE8BH+a+W+aE7+roT+Gh2BABThBBQG2hIApm7QmgQoPCt2sAx0MA5hAhWheBuhDeLhTsMBd+ZwwROBoRFB4RmhhBlE2GLKZerhLhiOtgMkGu-ydBpgDAIRKRk+Oh4RxR7hA2f+FRpG1RTeaRn63BuhbhpGk686AiPBKgfO-AHAg+OOeOEKBOYu4gPBrQYuP0JkSAg+pO5OAxUAQx1OHAIxuhLBLRfaP6H60Arhk6XBqx8RJRHhpK76LKLBNkFGsxlOVhDh++W+7G9hEwjh+B+xlRc606iRueaxaabRbxAi8aixg+3OvO-OSxNI+Q4xtkkx0xQuIuQJPRI+ohuBY8Rxv6WxPBnuOxpQGR-BzxpGGxJxPBZxzqe0jQcAsICAl40azROJfa3xBi7xle9RBGEK6RTRlBnxFiKGpmuutI7ANRfanJ+BTeG6W6zCEcbOg+eRvK+QkgrQAAamKUgBMVkFCTALrlAFKfCUkeydekeoYlAawOKUjBbjgNnmif-oiQIsiZsVAE4WfjUPUI0JwLsWaXyWmniX8k4brpblUFYBCkKNiUKbqfSR8YGf4NEgOvqYaUXFrJkRaSAFafiUkREfaV9E6ZiayVqdSW6UiUyomUmdJHgJtKEP6ZmaGb4sGWPlmRYsKaenUXBuRs6hmZWWIcxHhKPq6UQc7moK7uHg2OILaDDrVjeNYKwPAE3ojiABABQEAA)

### CODE REVIEW

**Comment by jjenzz** [comment]
I'm a big fan of this idea ❤️ . Thanks for the suggestion and work so far!

@colmtuite do we think it's worth continuing with the feature? Looks like it only needs:

- Apply a code theme that matches the one that was there before (unless we're happy with this one)
- Apply a max-height to the editor to match the previous max-height for the code block
- Render only one of the demos 😅

image

**Comment by jjenzz** [comment]
> sure we should render only one demo and improve style, it's just a minimal PoC to make it work, if you think it's legit, there are still a lot of works to do

Yep, I gathered as much, that's why I asked Colm if he thinks we should proceed. I mainly listed the remaining bits for his benefit so that he isn't confused by the two demos 😅

> BTW perhaps it's better to split the demo into [multi files](https://react-runner.vercel.app/#multi-files), like `index.tsx` and `styles.ts`, every time I find it's a bit hard to read the example 😅 

Love this idea too! @benoitgrelard and I had actually spoken about it recently.

**Comment by jjenzz** [comment]
@nihgwu i've spoken to folks internally and we're all super busy atm with various other priorities so won't be able to review/merge/help with this work in the short term, but we really like this idea so feel free to continue if you're happy that it might take some time.

**Comment by nihgwu** [comment]
@jjenzz OK

File: components/DocCodeBlock.tsx [hunk 1]
```diff
@@ -20,140 +25,161 @@ export function DocCodeBlock({
...
-
-        <Collapsible.Content forceMount>
-          <Box
-            css={{
-              position: 'relative',
-              ...(isCollapsed ? { display: 'none' } : {}),
-              ...(isCollapsible ? { top: '$2' } : { my: '$5' }),
-            }}
-          >
+    <>
```
Comment (nihgwu) [inline components/DocCodeBlock.tsx:39]: The change is very small actually, the wrapping of Fragment makes that
only two small additions, added `HeroContainer` and `CodeEditor` for live preview

**Comment by nihgwu** [comment]
@jjenzz I think it's ready for review, don't know why build failed

**Comment by andy-hook** [comment]
Hey @nihgwu !

Thanks so much for your work on this, and apologies for the delay. While it’s a really novel touch to be able to edit the primitives inline from the docs, as a team we’ve since been wondering if this would be that useful in the long term.

Given that CodeSandbox already offers such a comprehensive editing experience, and how common it is for us and the community to debug issues via CSB URLs, we’re not convinced this would offer much over the existing CSB external links. As a team we need to access value to users vs the effort required to improve and maintain features and so I hope this is understandable.

I’m going to close this now given the above, but thanks again for your effort and contribution on this, we really appreciate it 🙏

**Pull request closed without merging**